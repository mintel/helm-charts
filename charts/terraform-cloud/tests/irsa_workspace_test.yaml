suite: Test IRSA Workspaces
templates:
  - irsa-workspace.yaml
release:
  namespace: test-namespace
tests:
  - it: IRSA created for Dev S3 module workspace
    set:
      global.name: test-app
      global.clusterEnv: dev
      global.clusterName: cluster1
      global.clusterRegion: eu-west-1
      global.owner: sre
      global.partOf: test-project
      s3.enabled: true
    asserts:
      - isKind:
          of: Workspace
      - equal:
          path: metadata.namespace
          value: test-namespace
      - equal:
          path: metadata.name
          value: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      - equal:
          path: spec.module.source
          value: app.terraform.io/Mintel/app-iam/aws
      - hasDocuments:
          count: 1
      - equal:
          path: metadata.annotations.[app.mintel.com/altManifestFileSuffix]
          value: test-app-irsa
      - equal:
          path: metadata.annotations.[argocd.argoproj.io/sync-wave]
          value: "-20"
      - contains:
          path: spec.runTriggers
          content:
            sourceableName: dev-eu-west-1-cluster1-test-namespace-test-app-s3
      - contains:
          path: spec.variables
          content:
              environmentVariable: false
              hcl: false
              key: name
              sensitive: false
              value: test-app
      - contains:
          path: spec.variables
          content:
            environmentVariable: false
            hcl: false
            key: tfcloud_agent
            sensitive: false
            value: "true"
      - contains:
          path: spec.variables
          content:
            environmentVariable: false
            hcl: false
            key: eks_cluster_name
            sensitive: false
            value: cluster1
      - contains:
          path: spec.variables
          content:
            environmentVariable: false
            hcl: false
            key: aws_region
            sensitive: false
            value: eu-west-1
      - contains:
          path: spec.variables
          content:
            environmentVariable: false
            hcl: false
            key: aws_account_name
            sensitive: false
            value: dev
      - contains:
          path: spec.variables
          content:
            environmentVariable: false
            hcl: true
            key: tags
            sensitive: false
            value: |-
                {
                  Owner       = "sre"
                  Project     = "test-project"
                  Application = "test-app"
                }
