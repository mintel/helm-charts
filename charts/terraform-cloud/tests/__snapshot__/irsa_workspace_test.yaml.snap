IRSA created explicitly:
  1: |
    apiVersion: app.terraform.io/v1alpha1
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
        helm.sh/chart: terraform-cloud-0.38.0
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPoolID: ""
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.1.2
      omitNamespacePrefix: true
      organization: Mintel
      runTriggers: null
      secretsMountPath: /tmp/secrets
      sshKeyID: mintel-ssh
      terraformVersion: 1.0.7
      variables:
        - environmentVariable: false
          hcl: false
          key: aws_account_name
          sensitive: false
          value: dev
        - environmentVariable: false
          hcl: false
          key: aws_region
          sensitive: false
          value: eu-west-1
        - environmentVariable: false
          hcl: false
          key: eks_cluster_name
          sensitive: false
          value: cluster1
        - environmentVariable: false
          hcl: false
          key: k8s_namespace
          sensitive: false
          value: test-namespace
        - environmentVariable: false
          hcl: false
          key: k8s_service_account_name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: false
          key: name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: true
          key: tags
          sensitive: false
          value: |-
            {
              Owner       = "sre"
              Project     = "test-project"
              Application = "test-app"
            }
        - environmentVariable: false
          hcl: false
          key: tfcloud_agent
          sensitive: false
          value: "true"
IRSA created for Dev S3 module workspace:
  1: |
    apiVersion: app.terraform.io/v1alpha1
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-20"
        helm.sh/chart: terraform-cloud-0.38.0
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPoolID: ""
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.1.2
      omitNamespacePrefix: true
      organization: Mintel
      runTriggers:
        - sourceableName: dev-eu-west-1-cluster1-test-namespace-mntl-test-app-s3
      secretsMountPath: /tmp/secrets
      sshKeyID: mintel-ssh
      terraformVersion: 1.0.7
      variables:
        - environmentVariable: false
          hcl: false
          key: aws_account_name
          sensitive: false
          value: dev
        - environmentVariable: false
          hcl: false
          key: aws_region
          sensitive: false
          value: eu-west-1
        - environmentVariable: false
          hcl: false
          key: eks_cluster_name
          sensitive: false
          value: cluster1
        - environmentVariable: false
          hcl: false
          key: k8s_namespace
          sensitive: false
          value: test-namespace
        - environmentVariable: false
          hcl: false
          key: k8s_service_account_name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: false
          key: name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: true
          key: tags
          sensitive: false
          value: |-
            {
              Owner       = "sre"
              Project     = "test-project"
              Application = "test-app"
            }
        - environmentVariable: false
          hcl: false
          key: tfcloud_agent
          sensitive: false
          value: "true"
IRSA created with default (0) syncWave:
  1: |
    apiVersion: app.terraform.io/v1alpha1
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "0"
        helm.sh/chart: terraform-cloud-0.38.0
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPoolID: ""
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.1.2
      omitNamespacePrefix: true
      organization: Mintel
      runTriggers: null
      secretsMountPath: /tmp/secrets
      sshKeyID: mintel-ssh
      terraformVersion: 1.0.7
      variables:
        - environmentVariable: false
          hcl: false
          key: aws_account_name
          sensitive: false
          value: dev
        - environmentVariable: false
          hcl: false
          key: aws_region
          sensitive: false
          value: eu-west-1
        - environmentVariable: false
          hcl: false
          key: eks_cluster_name
          sensitive: false
          value: cluster1
        - environmentVariable: false
          hcl: false
          key: k8s_namespace
          sensitive: false
          value: test-namespace
        - environmentVariable: false
          hcl: false
          key: k8s_service_account_name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: false
          key: name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: true
          key: tags
          sensitive: false
          value: |-
            {
              Owner       = "sre"
              Project     = "test-project"
              Application = "test-app"
            }
        - environmentVariable: false
          hcl: false
          key: tfcloud_agent
          sensitive: false
          value: "true"
IRSA created with modified syncWave:
  1: |
    apiVersion: app.terraform.io/v1alpha1
    kind: Workspace
    metadata:
      annotations:
        app.mintel.com/altManifestFileSuffix: test-app-irsa
        app.mintel.com/terraform-allow-destroy: "true"
        app.mintel.com/terraform-cloud-tags: env:dev,owner:sre,mod:irsa
        app.mintel.com/terraform-owner: sre
        argocd.argoproj.io/sync-options: SkipDryRunOnMissingResource=true
        argocd.argoproj.io/sync-wave: "-100"
        helm.sh/chart: terraform-cloud-0.38.0
      labels:
        app.kubernetes.io/name: test-app-irsa
        app.mintel.com/env: dev
        app.mintel.com/owner: sre
        app.mintel.com/region: eu-west-1
        name: test-app-irsa
      name: dev-eu-west-1-cluster1-test-namespace-test-app-irsa
      namespace: test-namespace
    spec:
      agentPoolID: ""
      module:
        source: app.terraform.io/Mintel/app-iam/aws
        version: 2.1.2
      omitNamespacePrefix: true
      organization: Mintel
      runTriggers: null
      secretsMountPath: /tmp/secrets
      sshKeyID: mintel-ssh
      terraformVersion: 1.0.7
      variables:
        - environmentVariable: false
          hcl: false
          key: aws_account_name
          sensitive: false
          value: dev
        - environmentVariable: false
          hcl: false
          key: aws_region
          sensitive: false
          value: eu-west-1
        - environmentVariable: false
          hcl: false
          key: eks_cluster_name
          sensitive: false
          value: cluster1
        - environmentVariable: false
          hcl: false
          key: k8s_namespace
          sensitive: false
          value: test-namespace
        - environmentVariable: false
          hcl: false
          key: k8s_service_account_name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: false
          key: name
          sensitive: false
          value: test-app
        - environmentVariable: false
          hcl: true
          key: tags
          sensitive: false
          value: |-
            {
              Owner       = "sre"
              Project     = "test-project"
              Application = "test-app"
            }
        - environmentVariable: false
          hcl: false
          key: tfcloud_agent
          sensitive: false
          value: "true"
